### PR TITLE
Proposed affiliated package: pyds9plugin

### PR DESCRIPTION
**Pull request by vpicouet**
pyds9plugin is a plugin that pushes DS9 visualization software a step further, by allowing to analyze and process in real time images while keeping a high level of interactivity. The processing functions can then be generalized automatically to a set of images, to turn the quicklook tool into a real multi-processing pipeline. 

This plugin incorporates essential functions (radial profile,  2D/3D fitting, trough-focus, stacking, background removal,  source/artefact extraction, etc.) to extract quantitative and comprehensive information from imaging data sets in order to support instrumentation, reduce your observations, analyze the performance of your data, etc.
We also linked the most famous astronomical codes (SExtractor, SWARP, etc.) to the plugin to allow research-grade analysis and processing. 
As all researchers have different needs, the plugin is designed in a comprehensive way so that everyone can easily add its own macros that can then be run quickly and automatically on a set of images thanks to multi-processing.
This Pyds9plugin, available both on Pypi and Github (+ advertised in DS9 download page: https://sites.google.com/cfa.harvard.edu/saoimageds9/download), tries to gather a glimpse of all the possibilities that offers DS9 extensibility so that it motivates astronomers exploring  this implementation approach.

You can find a 3 minute video presentation here:
https://www.youtube.com/watch?v=XcDm2JQDMLY
and a draft/documentation where I also discuss the philosophy/goal of the project. I incorporated a comparison between DS9 features and similar astro/fits viewer softwares. 
https://nuage.osupytheas.fr/s/TW4dZsf5XzJcQFC

### CODE REVIEW

File: affiliated/registry.json [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+        "description": "A python DS9 extension for quicklook processing of astronomical images. This highly interactive extension can be generalized automatically to a set of images to turn the plug-in into a real multi-processing pipeline.",
+        "review": {
+         "functionality": "To be filled out by the reviewer",
+         "ecointegration": "To be filled out by the reviewer",
+         "documentation": "To be filled out by the reviewer",
+         "testing": "To be filled out by the reviewer",
+         "devstatus": "To be filled out by the reviewer",
+         "python3": "To be filled out by the reviewer",
+        "last-updated": "To be filled out by the reviewer"
+        }
```
Comment (pllim) [inline affiliated/registry.json:19]: This might fix the CI
```suggestion
```

**Comment by pllim** [comment]
Oh, no! I am so sorry! I misread the failure. The second commit is not needed (you need to drop it). The CI will pass when review happens. So sorry again!

Please be patient until someone from Astropy processes this request.

**Comment by vpicouet** [comment]
No problem @pllim, thanks for the info!

**Comment by hamogu** [comment]
Thank you for submitting pyds9plugin as a candidate for an affiliate
package for astropy. We have asked a reviewer from the Astropy project
for their opinion, and the review is generally positive. I also also
looked at pyds9plugin myself (as we usually do for affiliate packages)
and will give you our combined thoughts and ideas below.

However, ultimately we, the affiliate package editors, feel that
pyds9plugin is out of scope for an astropy-affiliated package. That is
no statement of code quality or usefulness (I will probably use it for
my own science in the future), but pyds9plugin is fundamentally a plugin
for ds9, not for astropy. It makes a lot of astropy functionality
available to users of ds9, but we think it falls more into a category
similar to, e.g. jupyter notebooks. It provides a tool or an interface
that can be used to run astropy-related functionality but we think it
would be more appropriately called "ds9-affiliated". (I know that ds9
does not use that term, but, as you point out, they do list your package
on the ds9 website). It's difficult to draw a hard line what exactly the
scope of astropy affiliated packages is and consequently I have to admit
that the description on the website is a little vague - something we
hope to improve on in the future.

While this decision is not based on the content of the review, we want
to just list a few ideas or suggestions that came up in the process:

### Some notes on installation:

+ Installation worked as advertised with pip. Currently installation
does require a compiler. There are a few informational messages flashed
in green during installation about trying out `pyds9plugin` -- these
went by so quickly it would have been easy to miss them entirely. I'd
recommend dropping them because they are likely to be missed by users.
+ Activating the plugin was confusing. Adding some more text to the
documentation explaining that one will need to do some copy/paste of
paths to get the extension autoloading would be helpful.
+ Advise folks on a Mac that they should switch to "Light Mode" theme
before running ds9 -- in dark theme its text on buttons, etc., is
unreadable. This isn't really a `pyds9plugin` issue but it seems better
to address it to reduce user frustration.

### Other comments

The functionality is primarily accessed through the ds9 UI, something I
didn't fully understand at first.

## Integration with Astropy ecosystem

The package uses a mix of astropy packages (e.g. `photutils` in a couple
of places) but also uses a few of the AstroMatic codes, `SExtractor` and
`SWARP` at a minimum. I'd encourage the author to consider using `sep`
in place of `SExtractor` for ease of installation. While `sep` is not an
Astropy affiliated package, it is more Python and offers a Python API to
`SExtractor`. It might be possible use `reproject` where `SWARP` is used.

It is fine to keep both `SExtraxtor` and `SWARP` as *options* for users
-- it would simply be nice to have `pip`-installable options too to
simplify installation for as many users as you can.

Some of the header-indexing code is already available in
`ccdproc.ImageFileCollection` though there may be reasons the author has
chosen to do their own implementation.

## Documentation

We really like the frequent videos, though we'd recommend finding a way
to cloud host them so users do not have to download a bunch `.mov` files
to watch them - actually, some are on YouTube, but some are not.

We really, really like the tutorial accessible through ds9 after the
plugin is installed -- please move that much, much earlier in the
documentation! It was especially helpful because I am not at all
familiar with ds9.

The documentation provided is about 95% writing about viewers and image
processors in general in astronomy and about 5% about how to use
`pyds9plugin`.

The *documentation* portion should be separated from what feels like a
draft of an academic paper. I would suggest adding a few "getting
started"-style examples that would allow a user unfamiliar with ds9 to
be able to try out some of the features of `pyds9plugin`.

A screenshot showing where to find the `pyds9plugin` functions would be
helpful.

## Testing
Though there are two test scripts in `pyds9plugin/testing` each has
hardcoded paths to the author's directories, but of course testing
something that's supposed to be driven by users clicking on a buttons in
ds9 can't really be tested in the same way we test other Python packages.

## Other notes/comments

+ The repository is **0.67 GB**! That is....huge.

<table>
<tr><td><b>Functionality/Scope</b></td><td>
<a href="https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md#functionality-functionality">
Out of scope
</a></td></tr>

<tr><td colspan=2><i>See full text above</i></td></tr>

<tr><td><b>Integration with Astropy ecosystem</b></td><td>
<a href="https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md#integration-with-astropy-ecosystem--ecointegration">
Orange
</a></td></tr>

<tr><td colspan=2><i>See full text above</i></td></tr>

<tr><td><b>Documentation</b></td><td>
<a href="https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md#documentation-documentation">
Orange
</a></td></tr>

<tr><td colspan=2><i>Good, but a few suggestions above.</i></td></tr>

<tr><td><b>Testing</b></td><td>
<a href="https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md#testing-testing">
Orange
</a></td></tr>

<tr><td colspan=2><i>We understand that automatic testing is is not really possible for this project, so this category is not fully applicable.</i></td></tr>

<tr><td><b>Development status</b></td><td>
<a href="https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md#development-status-devstatus">
Heavy Development
</a></td></tr>
<tr><td colspan=2><i>The repository itself has frequent activity. We would have marked this green except that the only releases on PyPI are listed as dev: `3.1.dev5` and `3.1.dev7`.</i></td></tr>

<tr><td><b>Python 3 compatibility</b></td><td>
<a href="https://github.com/astropy/astropy-procedures/blob/main/documents/affiliated_package_review_guidelines.md#python-3-compatibility-python3">
Green
</a></td></tr>

<tr><td colspan=2><i>No further comments</i></td></tr>

</table>

As per the review guidelines, we
therefore won't be able to accept this package as an affiliated package.
We will leave this pull request open for a month in case you would like to
respond to the comments and/or address any of them.

If you have any follow-up questions or disagree with any of the comments above,
leave a comment and we can discuss it here. At any point in future you can
request a re-review of the package if you believe any of the scores should be
updated - contact the affiliated package editors
[listed on the Astropy team page](https://www.astropy.org/team),
and we will do a new review.

**Pull request closed without merging**